EZSwiftExtensionsTests/DateTests.swift

```diff
@@ -11,46 +11,58 @@ import XCTest
 
-    var string: String!
-    var wrongDateString: String!
-    let format = "dd-MM-yyyy hh:mm:ss"
+    var dateString: String!
+    var invalidDateString: String!
+    var date: Date!
+    var timeZoneOffset: TimeInterval!
+    let format = "yyyy-MM-dd'T'HH:mm:ssxxxxx"
 
     override func setUp() {
         super.setUp()
-        string = "01-01-1970 00:00:00"
-        wrongDateString = "13-82-1900 90:65:12"
+        self.dateString = "2015-11-14T16:05:11+00:00" // time of the first commit of EZSwiftExtension :clap::clap:
+        self.invalidDateString = "9999-99-99T99:99:99+00:00"
+        self.date = Date(fromString: self.dateString, format: self.format)!
+        NSTimeZone.default = TimeZone(abbreviation: "SGT")! // set timezone to be SGT
+        self.timeZoneOffset = TimeInterval(NSTimeZone.local.secondsFromGMT())
     }
 
     func testDateFromString() {
-        guard let dateFromString = Date(fromString: string, format: format) else {
+        guard let dateFromString = Date(fromString: self.dateString, format: self.format) else {
             XCTFail("Date From String Couldn't be initialized.")
             return
         }
-        let formatter = DateFormatter()
-        formatter.dateFormat = format
-        let dateString = formatter.date(from: string)
-        XCTAssertEqual(dateFromString, dateString)
-        XCTAssertNil(Date(fromString: wrongDateString, format: format), "Date From String initialized, but source string was invalid.")
+        
+        NSTimeZone.default = TimeZone(abbreviation: "UTC")! // set timezone to be UTC to match with original string
+        XCTAssertEqual(dateFromString.toString(format: self.format), self.dateString!) // TODO why is there a need for ! for self.dateString
+        XCTAssertNil(Date(fromString: self.invalidDateString, format: format), "Date From String initialized, but source string was invalid.")
         
         let dateFromFalseStr = Date(fromString: "lol", format: "haha")
         XCTAssertNil(dateFromFalseStr)
     }
 
     func testHTTPDateString() {
         // Given
-        let expectedResult = Date(timeIntervalSince1970: 784_111_777)
-        let fromGMT = TimeInterval(NSTimeZone.local.secondsFromGMT())
+        let fromStartOfDay = TimeInterval(16 * 3600 + 5 * 60 + 11) // seconds from start of day
+        let fromStartOfMin = TimeInterval(11) // seconds from start of minute
 
         // When
-        let rfc1123 = Date(httpDateString: "Sun, 06 Nov 1994 08:49:37 GMT")
-        let rfc850 = Date(httpDateString: "Sunday, 06-Nov-94 08:49:37 GMT")
-        let asctime = Date(httpDateString: "Sun Nov  6 08:49:37 1994")
-        let invalid = Date(httpDateString: "1994-11-06 08:49:37")
+        let rfc1123 = Date(httpDateString: "Sat, 14 Nov 2015 16:05:11 GMT")
+        let rfc850 = Date(httpDateString: "Saturday, 14-Nov-15 16:05:11 GMT")
+        let asctime = Date(httpDateString: "Sun Nov 14 16:05:11 2015")
+        let iso8601DateOnly = Date(httpDateString: "2015-11-14")
```

Comment:
It is to simulate a json string, and it needs to match the base `dateString`.

Do you mean we should put this string into a variable and and then append the remaining parts of the date string as necessary?